test/style/conversion/raster_dem_options.test.cpp

```diff
@@ -0,0 +1,47 @@
+#include <mbgl/test/util.hpp>
+
+#include <mbgl/style/conversion/json.hpp>
+#include <mbgl/style/conversion/raster_dem_options.hpp>
+#include <mbgl/util/tileset.hpp>
+
+#include <mbgl/util/logging.hpp>
+
+using namespace mbgl;
+using namespace mbgl::style;
+using namespace mbgl::style::conversion;
+
+TEST(RasterDEMOptions, Basic) {
+    Error error;
+    std::optional<RasterDEMOptions> converted = convertJSON<RasterDEMOptions>("{}", error);
+    ASSERT_TRUE((bool)converted);
+}
+
+TEST(RasterDEMOptions, ErrorHandling) {
+    Error error;
+    std::optional<RasterDEMOptions> converted = convertJSON<RasterDEMOptions>(
+        R"JSON({
+        "encoding": "this isn't a valid encoding"
+    })JSON",
+        error);
+    ASSERT_FALSE((bool)converted);
```

Comment:
Shouldn't need a cast.

Also you can assert the error message here I think.

Reply:
I was following the pattern as established [here](https://github.com/maplibre/maplibre-native/blob/main/test/style/conversion/geojson_options.test.cpp), but happy to do something different if you think it warrants it.